web/client/epics/maps.js

```diff
@@ -0,0 +1,255 @@
+} = require('../selectors/maps');
+const {
+    mapIdSelector, mapInfoDetailsUriFromIdSelector
+} = require('../selectors/map');
+const {
+    currentMapDetailsTextSelector, currentMapIdSelector,
+    currentMapDetailsUriSelector, currentMapSelector,
+    currentMapDetailsChangedSelector, currentMapOriginalDetailsTextSelector
+} = require('../selectors/currentmap');
+const {
+    currentMessagesSelector
+} = require('../selectors/locale');
+
+const {userParamsSelector} = require('../selectors/security');
+const {manageMapResource, deleteResourceById, getIdFromUri} = require('../utils/ObservableUtils');
+const ConfigUtils = require('../utils/ConfigUtils');
+
+/**
+    If details are changed from the original ones then set unsavedChanges to true
+*/
+const setDetailsChangedEpic = (action$, store) =>
+    action$.ofType(SAVE_DETAILS)
+    .switchMap((a) => {
+        let actions = [];
+        const state = store.getState();
+        const detailsUri = currentMapDetailsUriSelector(state);
+        if (a.detailsText.length <= 500000) {
+            actions.push(toggleDetailsSheet(true));
+        } else {
+            actions.push(basicError({message: "maps.feedback.errorSizeExceeded"}));
+        }
+        if (!detailsUri) {
+            actions.push(setDetailsChanged(a.detailsText !== "<p><br></p>"));
+            return Rx.Observable.from(actions);
+        }
+        const originalDetails = currentMapOriginalDetailsTextSelector(state);
+        const currentDetails = currentMapDetailsTextSelector(state);
+        actions.push(setDetailsChanged(originalDetails !== currentDetails));
+        return Rx.Observable.from(actions);
+    });
+
+
+/**
+ * If the details resource does not exist it saves it, and it updates its permission with the one set for the mapPermissionsFromIdSelector
+ * and it updates the attribute details in map resource
+*/
+const saveResourceDetailsEpic = (action$, store) =>
+    action$.ofType(SAVE_RESOURCE_DETAILS)
+    .switchMap(() => {
+        const state = store.getState();
+        const mapId = currentMapIdSelector(state);
+        const value = currentMapDetailsTextSelector(state, mapId);
+        const detailsChanged = currentMapDetailsChangedSelector(state);
+
+        let params = {
+            attribute: "details",
+            map: currentMapSelector(state),
+            resource: null,
+            type: "STRING"
+        };
+        if (!detailsChanged) {
+            return Rx.Observable.of(doNothing());
+        }
+        if (value !== "" && detailsChanged) {
+            params.resource = {
+                category: "DETAILS",
+                userParams: userParamsSelector(state),
+                metadata: {name: uuidv1()},
+                value,
+                permissions: mapPermissionsFromIdSelector(state, mapId),
+                optionsAttr: {},
+                optionsRes: {}
+            };
+            params.messages = currentMessagesSelector(state);
```

Comment:
Notification system already provide localization. You don't need any access to localization system. Remove all unnecessary code for notifications and use message IDs.

Reply:
Notification system doesn't support custom params just localized strings.
becasue internally they use [getMessageById](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/notifications/NotificationContainer.jsx#L83)
 as you can see i'm adding the name of the map in the notification message which i can simply remove.  suggestions?

Reply:
remove name for now (and all message retrivial code)